force-app/main/default/classes/BDI_MatchDonations.cls

```diff
@@ -513,7 +513,7 @@ public with sharing class BDI_MatchDonations implements BDI_IMatchDonations {
             strSoql += ' ' + UTIL_Namespace.StrTokenNSPrefix('Primary_Contact__c') + ' IN :setConId ';
             cFilter++;
         }
-        strSoql += ') ';
+        strSoql += ') LIMIT 50000 ';
```

Comment:
How about extracting lines 490-516 into a "buildQuery" method, so that it can be made @TestVisible and enable a test assertion that the LIMIT is now included in the query string?

Reply:
Done!